src/Pelagos/Bundle/AppBundle/Resources/public/js/entity/ResearchGroup.js

```diff
@@ -91,28 +93,25 @@ $(document).ready(function()
  */
 function addOptionsByEntity(selectElement, entity, filter)
 {
-    "use strict";
-
-    var url = $(selectElement).attr("data-url");
-    if (typeof filter !== "undefined") {
-        url += "?" + filter;// + "&properties=id,name";
-    } else {
-        //url += "?properties=id,name";
-    }
+    return $.Deferred(function() {
+        var url = $(selectElement).attr("data-url");
+        if (typeof filter !== "undefined") {
+            url += "?" + filter;// + "&properties=id,name";
+        } else {
+            //url += "?properties=id,name";
+        }
 
-    $.ajax({
-        url: url,
-        dataType: "json",
-        async: false
-    })
-    .done(function(data) {
-        var entities = sortObject(data, "name", false, true);
+        $.getJSON(url, function(data) {
+             var entities = sortObject(data, "name", false, true);
 
-        $.each(entities, function(seq, item) {
-            selectElement.append(
-                $("<option></option>").val(item.id).html(item.name)
-            );
+            $.each(entities, function(seq, item) {
+                selectElement.append(
+                    $("<option></option>").val(item.id).html(item.name)
+                );
+            });
+            //selectElement.find("option[value=\"" + selectElement.attr(entity) + "\"]").attr("selected", true);
         });
-        //selectElement.find("option[value=\"" + selectElement.attr(entity) + "\"]").attr("selected", true);
+        
+        return this.resolve();
```

Comment:
instead of this, it seems to work fine with chaining .promise() to the end of the outer return. Why the 2 returns?

Reply:
Second return wasn't needed, removed it.